Automattic/jetpack
# My Jetpack: Re-enable My Jetpack module-product tests

Fixes #26219.

After merging #26218, that fixed issues with the WorDBless setup, we identified that some tests were broken and we didn't detect before. That is the case for the My Jetpack `test-module-product.php` suite. We activelly skipped this to move the WorDBless fix forward, but now we are bringing it back.

The `test-module-product.php` suite depends on a mocked Jetpack plugin/class version, so we need to update this mocked class to run the tests properly.

It also needs a concrete implementation of `Module_Product` class, so I added a `Sample_Module_Product` to use as the subject of the tests, moving away from the use of real product classes as `Videopress` (which is not a `Module_Product` since #25954 and therefore would break the tests anyway).

#### Changes proposed in this Pull Request:

* Remove the `markTestSkipped` call from the test `set_up`
* Update the Jetpack class mock to add the method `is_connection_ready`, necessary for one of the operations tested
* Update the Jetpack class mock to add constants required by `Automattic\Jetpack\Modules` class when we have a real Jetpack installation present:
   * `JETPACK__VERSION`, set to `0.0.0` just so we have some value
   * `JETPACK_MASTER_USER` using the same value as the real plugin
   * `JETPACK__API_VERSION`, idem
   * `JETPACK__PLUGIN_DIR`, idem
   * `JETPACK__PLUGIN_FILE`, idem
* Create `Sample_Module_Product` class to use as the test subject, extending `Module_Product` and implementing just the minimum required code to become concrete
* Update the test suite to rely on the new `Sample_Module_Product` class instead of using `Videopress` or other concrete products

#### Other information:


#### Jetpack product discussion



None.

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Make sure you installed/updated all the dependencies for the project, using `jetpack install --all`
* Go to `projects/packages/my-jetpack`
* Run `composer phpunit`

All the tests should pass.
